### ISSUE TITLE
Some improvements to documentation

### ISSUE THREAD

**Issue by AlanObject** [ckeditor/ckeditor5-angular#61]
The current document leaves some questions unanswered and could use some examples.  I'll list a couple of topics that could be addressed.

1. It would be good to have a topic of how to use the CKEditor API from an Angular application.  On my own I figured out how to access **CKEditorComponent.editorInstance** but it would have been nice to be guided to that.

2. The role and description of the **typings.d.ts** file doesn't seem to be mentioned anywhere.   I only learned about it by dissecting the sample application, comparing file by file.  Also: the file I imported from there uses the **var** keyword instead of **const**.

3. I use npm and I had to copy the ckeditor component/module into my project to get it to work.   It doesn't work if you try to reference it in the **node_modules** structure.  

4. This is probably not a documentation issue but I thought I would mention it here.  If you look at **ckeditor.component.ts** in WebStorm and ask it to reformat the code -- _the code breaks_.  You end up with a component that compiles but does not properly bind to the **ngModel**.   Sorry I haven't had time to troubleshoot this issue but maybe a warning in release notes would be good.  

If someone is willing to guide me in how to contribute I would be willing to update the documentation.

**Comment by ma2ciek** [comment]
Hi, @AlanObject!

Thanks for the feedback :)

> It would be good to have a topic of how to use the CKEditor API from an Angular application. On my own I figured out how to access CKEditorComponent.editorInstance but it would have been nice to be guided to that.

We didn't document it because it's rather not a frequent use case. We documented the common one - how to get the editor instance from your own component that operates on the `<ckeditor></ckeditor>` - https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/angular.html#ready. But we'll be happy to extend that part.

> The role and description of the typings.d.ts file doesn't seem to be mentioned anywhere. I only learned about it by dissecting the sample application, comparing file by file. Also: the file I imported from there uses the var keyword instead of const.

Yep, my apologies. I forgot to note about that file in the readme. Recently we're focused on the TS (and typings) topic - https://github.com/ckeditor/ckeditor5/issues/1415, https://github.com/ckeditor/ckeditor5/issues/504. That would cover much wider this topic but for now it's a no-go for us, unfortunately, because of the huge cost of migration. But we'll be investigating some other ways to provide good typings for the CKEditor 5 in the future.

**Comment by ma2ciek** [comment]
>>  The role and description of the typings.d.ts file doesn't seem to be mentioned anywhere. I only learned about it by dissecting the sample application, comparing file by file. Also: the file I imported from there uses the var keyword instead of const.

> Yep, my apologies. I forgot to note about that file in the readme. Recently we're focused on the TS (and typings) topic - ckeditor/ckeditor5#1415, ckeditor/ckeditor5#504. That would cover much wider this topic but for now it's a no-go for us, unfortunately, because of the huge cost of migration. But we'll be investigating some other ways to provide good typings for the CKEditor 5 in the future.

Oh, my mistake. I was talking about the `ckeditor.ts` file. I assume that you're using the `strict` mode?

**Comment by ma2ciek** [comment]
> If someone is willing to guide me in how to contribute I would be willing to update the documentation.

If you want to contribute to the developer guide then we'd be happy to see your PR in the https://github.com/ckeditor/ckeditor5 repository (this [file](https://github.com/ckeditor/ckeditor5/blob/master/docs/builds/guides/frameworks/angular.md) contain the documentation for the angular integration) :)

**Comment by oleq** [comment]
You can check out the [contributors guide](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/contributing.html) too.

**Comment by ma2ciek** [comment]
And to clarify things about the `typings.d.ts` - that file can't be exported by the `ckeditor5-angular` package because these typings are for CKEditor 5 builds and we can't guess which build actually a user will be using. We can only add information to the README, how to add these typings by yourself if you use the strict mode.

**Comment by ma2ciek** [comment]
> I would suggest exporting a file that had several different options included but all commented out. The user then simply needs to uncomment the variant they are using. Instructions can be included there to that effect. (A common technique in text configuration files you find in Linux/BSD packages.)

You should be able to build your angular app using `ckeditor5-angular` as NPM dependency. Which means that you can't force anybody to edit any file inside `node_modules`. Cloning this repo shouldn't be the recommended approach unless you're creating a PoC.

**Comment by ma2ciek** [comment]
And could you tell how importing the `ckeditor5-angular` package resulted in the error? Since all your other issues rely on that.

**Comment by AlanObject** [comment]
For a documentation PR do I need to create an issue in ckeditor/ckeditor5 repository and name a branch after that?

**Comment by ma2ciek** [comment]
You can just create a branch named `t/ckeditor5-angular/61` in the main repository (`ckeditor/ckeditor5`). This is our convention.

Note that I've made some changes recently in exported members: https://github.com/ckeditor/ckeditor5-angular/issues/66.

**Comment by AlanObject** [comment]
OK I did a fork/branch/edit/PR but I am not sure what to do next.  

New work is [here](https://github.com/AlanObject/ckeditor5/blob/t/ckeditor5-angular/61/docs/builds/guides/frameworks/angular.md).

**Comment by ma2ciek** [comment]
Thanks for the PR. I'll review it soon.

But could you also answer the questions I've asked here: https://github.com/ckeditor/ckeditor5-angular/issues/61#issuecomment-459277285. Especially:

1. What error have you been facing while you were using NPM?
2. Why you were forced to use `CKEditorComponent.editorInstance`? Did you consider accessing the editor via `<ckeditor (ready)="onReady($event)">` and `Component#onReady( editor ){}`?

**Comment by AlanObject** [comment]
1. I will look into this the npm issue later today and try to re-create the problem.  I ended up copying the component source into the src folder because that's what the demo program does.   Once that started working I left it that way and didn't try to diagnose the root cause.  

2. I didn't consider using the (ready) event mostly because I didn't think of it and I didn't see an example where it was used to access the object.   What I wanted to do is manipulate the model from the component when the user clicked a button and not just when the editor reached the ready lifecycle phase.   So the template variable approach seemed most convenient.   However I am now thinking that both methods should be documented.

**Comment by AlanObject** [comment]
> 1. What error have you been facing while you were using NPM?

I have instantiated a new project to test this and I have been unable to reproduce the problem.  In the process I have discovered a number of current version compatibility problems* that had to be worked around but once I did the editor was easily imported from the `node-modules` as expected.

At this point I am going to withdraw the text in my PR that speaks to this.

* -- this has nothing to do with CKEditor but if you are interested it turns out that RXJS current version does not work with the following combination:

```
